src/converter/jfr2pprof.java

```diff
@@ -208,20 +244,28 @@ private byte[] getMethodName(final long methodId, final byte methodType) {
         }
     }
 
-    public static void main(String[] args) throws Exception {
-        if (args.length < 2) {
-            System.out.println("Usage: java " + jfr2pprof.class.getName() + " input.jfr output.pprof");
+    public static void main(String[] cmdline) throws Exception {
+        Arguments args = new Arguments(cmdline);
+        if (args.input == null) {
+            System.out.println("Usage: java " + jfr2pprof.class.getName() + " [options] input.jfr [output.pprof]");
+            System.out.println();
+            System.out.println("options support one of the following events:");
+            System.out.println("  --cpu         CPU events (default)");
+            System.out.println("  --alloc       Allocation events");
             System.exit(1);
         }
 
-        File dst = new File(args[1]);
+        Class<? extends Event> eventClass = args.alloc ? AllocationSample.class : ExecutionSample.class;
+
+        String output = Optional.ofNullable(args.output).orElse(args.input.replace(".jfr", ".pprof"));
```

Comment:
I'd rewrite this with a regular `if`: `Optional` makes this harder to follow. Also, `args.input.replace` will be called regardless of the value of `output`.